### PR TITLE
[docs][KubeRay] Provide some GKE instructions in KubeRay example

### PR DESCRIPTION
**Pull request by kevin85421**
## Why are these changes needed?

* `ml-example.md`: I used a GKE cluster without autopilot in this example. As there are some dependency issues on my Mac M1 at this moment, I made some slight modifications to the reproduction instructions. Instead of running the job locally, I used kubectl exec to log in to the head Pod and submit the XGBoost job. This change should not have impact on this document.

  Screen Shot 2023-04-10 at 3 34 47 PM
  Screen Shot 2023-04-10 at 3 34 37 PM

* gpu-training-example.md
  * Protobuf issue (`ray-ml:2.3.0-gpu`): https://github.com/ray-project/ray/issues/31309#issuecomment-1407226449 => Choose a Ray 2.2 image.
  * `TorchVisionPreprocessor` (Ray 2.2 does not support TorchVisionPreprocessor. Hence, I used `pytorch_training_e2e.py` in the branch `ray-2.2.0`) 
    https://github.com/ray-project/ray/blob/0a8471e428db203e4905428cf2566a4a1dbfdab5/release/air_tests/air_benchmarks/workloads/pytorch_training_e2e.py#L15
    Screen Shot 2023-04-10 at 5 46 50 PM

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by kevin85421** [comment]
cc @gvspraveen

File: doc/source/cluster/kubernetes/examples/ml-example.md [hunk 1]
```diff
@@ -14,11 +14,36 @@ workload on Kubernetes infrastructure.
...
+node group or pool with a 1 node minimum and a 10 node maximum.
+The 1 static node will be used to run the Ray head pod. This node may also host the KubeRay
+operator and Kubernetes system components. After the workload is submitted, 9 additional nodes will
+scale up to accommodate Ray worker pods. These nodes will scale back down after the workload is complete.
+```
 
 If you are new to Kubernetes and you are planning to deploy Ray workloads on a managed
-Kubernetes service, we recommend taking a look at this {ref}`introductory guide <kuberay-k8s-setup>`
-first.
+Kubernetes service on other cloud providers, we recommend taking a look at this {ref}`introductory guide <kuberay-k8s-setup>` first.
```
Comment (architkulkarni) [inline doc/source/cluster/kubernetes/examples/ml-example.md:46]: Should we move this paragraph near the top of the section like it was before?

File: doc/source/cluster/kubernetes/examples/ml-example.md [hunk 2]
```diff
@@ -14,11 +14,36 @@ workload on Kubernetes infrastructure.
...
+```
+
+Make sure you are connected to your Kubernetes cluster. For GCP, you can do so by:
+* Navigate to your GKE cluster page, and click "CONNECT" button. Then, copy "Command-line access".
+* `gcloud container clusters get-credentials <your-cluster-name> --region <your-region> --project <your-project>` ([Link](https://cloud.google.com/sdk/gcloud/reference/container/clusters/get-credentials))
+* `kubectl config use-context` ([Link](https://kubernetes.io/docs/tasks/access-application-cluster/configure-access-multiple-clusters/))
+
+```{admonition} Optional: Set up an autopilot GKE cluster
+**If you would like to try running the workload with autoscaling enabled**, use an autoscaling
+node group or pool with a 1 node minimum and a 10 node maximum.
```
Comment (architkulkarni) [inline doc/source/cluster/kubernetes/examples/ml-example.md:39]: Is this the same as `--num-nodes=1 --min-nodes 1 --max-nodes 10 --enable-autoscaling` as written above? Writing a more specific command or isntruction here would be ideal.

**Review by architkulkarni — APPROVED** [review]
Looks good! Comments non-blocking

File: doc/source/cluster/kubernetes/examples/ml-example.md (hunk 2 above)
Comment (kevin85421) [reply doc/source/cluster/kubernetes/examples/ml-example.md]: Yes, I moved this comment to the `gcloud` command.

https://github.com/ray-project/ray/pull/33339/commits/362b7983edc37a7255bbb50154d2cb72d2068db3

File: doc/source/cluster/kubernetes/examples/ml-example.md (hunk 1 above)
Comment (kevin85421) [reply doc/source/cluster/kubernetes/examples/ml-example.md]: Done

**Comment by kevin85421** [comment]
> @kevin85421 let me know when this ready to merge!

It is ready to review!

**Comment by architkulkarni** [comment]
Windows test_serve_integration unrelated and was flaky on master
Serve gradio tests unrelated and were flaky on master
Lint failure unrelated
Linkcheck failure unrelated

**Review by architkulkarni — APPROVED** [review]

**Comment by architkulkarni** [comment]
@kevin85421 Shall I merge this?

**Comment by kevin85421** [comment]
> @kevin85421 Shall I merge this?

Yes, this is the last step for KubeRay release. Thank you!

**Pull request merged**